src/popup/components/ErrorKeyRevoked.tsx

```diff
@@ -0,0 +1,192 @@
+import React from 'react'
+import { useForm } from 'react-hook-form'
+import { AnimatePresence, m } from 'framer-motion'
+import { WarningSign } from '@/popup/components/Icons'
+import { Button } from '@/popup/components/ui/Button'
+import { Code } from '@/popup/components/ui/Code'
+import { useTranslation } from '@/popup/lib/context'
+import { useLocalStorage } from '@/popup/lib/hooks'
+import type { PopupStore } from '@/shared/types'
+import type { Response } from '@/shared/messages'
+
+interface Props {
+  info: Pick<PopupStore, 'publicKey' | 'walletAddress'>
+  disconnectWallet: () => Promise<Response>
+  reconnectWallet: () => Promise<Response>
+  onReconnect?: () => void
+  onDisconnect?: () => void
+}
+
+type Screen = 'main' | 'reconnect'
+
+export const ErrorKeyRevoked = ({
+  info,
+  disconnectWallet,
+  reconnectWallet,
+  onReconnect,
+  onDisconnect
+}: Props) => {
+  const [screen, setScreen, clearScreen] = useLocalStorage<Screen>(
+    'keyRevokedScreen',
+    'main',
+    { maxAge: 2 * 60 }
+  )
+
+  if (screen === 'main') {
+    return (
+      <AnimatePresence mode="sync">
+        <MainScreen
+          requestReconnect={() => setScreen('reconnect')}
+          disconnectWallet={disconnectWallet}
+          onDisconnect={onDisconnect}
+        />
+      </AnimatePresence>
+    )
+  } else {
+    return (
+      <AnimatePresence mode="sync">
+        <ReconnectScreen
+          info={info}
+          reconnectWallet={reconnectWallet}
+          onReconnect={() => {
+            clearScreen()
+            onReconnect?.()
+          }}
+        />
+      </AnimatePresence>
+    )
+  }
+}
+
+interface MainScreenProps {
+  disconnectWallet: Props['disconnectWallet']
+  onDisconnect?: Props['onDisconnect']
+  requestReconnect: () => void
+}
+
+const MainScreen = ({
+  disconnectWallet,
+  onDisconnect,
+  requestReconnect
+}: MainScreenProps) => {
+  const t = useTranslation()
+  const [errorMsg, setErrorMsg] = React.useState('')
+  const [loading, setIsLoading] = React.useState(false)
+
+  const requestDisconnect = async () => {
+    setErrorMsg('')
+    try {
+      setIsLoading(true)
+      await disconnectWallet()
+      onDisconnect?.()
+    } catch (error) {
+      setErrorMsg(error.message)
+    }
+    setIsLoading(false)
+  }
+
+  return (
+    <m.div
+      initial={{ opacity: 0 }}
+      animate={{ opacity: 1 }}
+      exit={{ opacity: 0 }}
+      className="space-y-4 text-sm"
+    >
```

Comment:
Not sure if it's only me, but I find this a little bit too flashy? I'm okay with keeping the animation, but only when switching between states and remove the animation on first render.